ethers-providers/src/transports/quorum.rs

```diff
@@ -160,23 +160,41 @@ impl<T> QuorumProviderBuilder<T> {
 }
 
 impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
-    /// Returns the block height that _all_ providers have surpassed.
-    ///
-    /// This is the minimum of all provider's block numbers
-    async fn get_minimum_block_number(&self) -> Result<U64, ProviderError> {
-        let mut numbers = join_all(self.providers.iter().map(|provider| async move {
+    /// Returns the block height that a _quorum_ of providers have reached.
+    async fn get_quorum_block_number(&self) -> Result<U64, QuorumError> {
+        let (oks, errs) = join_all(self.providers.iter().map(|provider| async move {
             let block = provider.inner.request("eth_blockNumber", QuorumParams::Zst).await?;
-            serde_json::from_value::<U64>(block).map_err(ProviderError::from)
+            serde_json::from_value::<U64>(block).map(|b| (provider, b)).map_err(ProviderError::from)
         }))
         .await
         .into_iter()
-        .collect::<Result<Vec<_>, _>>()?;
-        numbers.sort();
-
-        numbers
-            .into_iter()
-            .next()
-            .ok_or_else(|| ProviderError::CustomError("No Providers".to_string()))
+        .partition::<Vec<Result<(&WeightedProvider<T>, U64), ProviderError>>, _>(|res| res.is_ok());
+
+        let mut numbers = oks.into_iter().map(Result::unwrap).collect::<Vec<_>>();
+
+        numbers.sort_by(|(_, block_a), (_, block_b)| {
+            // order by descending block number
+            block_a.cmp(block_b).reverse()
+        });
+
+        // find the highest possible block number a quorum agrees on
+        let mut weight_set = vec![];
+        let mut block = U64::from(0);
+        for (provider, n) in numbers.iter().copied() {
+            weight_set.push(provider);
+            debug_assert!(block == U64::from(0) || block >= n);
```

Comment:
Curious why the `block == U64::from(0)` condition? I don't think I'd ever expect that to happen

Reply:
We initialize it to zero.

Reply:
lmao oops sorry